deploy/resources.go

```diff
@@ -2797,6 +2797,29 @@ artifacts:
       passive: false
     version: 2.23.1
 
+`
+	Resources["cr-example.yaml"] =
+		`
+apiVersion: camel.apache.org/v1alpha1
+kind: Integration
+metadata:
+  name: example
+spec:
+  source:
+    content: |-
+      // This is Camel K Groovy example route
+
+      rnd = new Random()
+
+      from('timer:groovy?period=1s')
+          .routeId('groovy')
+          .setBody()
+              .constant('Hello Camel K!')
+          .process {
+              it.in.headers['RandomValue'] = rnd.nextInt()
+          }
+          .to('log:info?showHeaders=true')
+    name: routes.groovy
```

Comment:
May this be changed by the fact that your pr is created from an old commit ?

Reply:
Not sure. I added a sorting option to `ls` when creating that file previously, but it seems that the PR has not been taken over. Actually it then depends on the way how ls works on different systems how resources are picked up. My ls uses alphabetical sort order by default, so orders 'cr-example.yml' after 'camel-catalog-2.23.1.yaml' and before 'crd-camel-catalog.yaml'.

In this file frommaster 'cr-example.yml' comes right before 'operator-deployment-kubernetes.yml' which seems wrong if you assume alphabetical ordering.

I'd recommend to add a sorting option to ls to not depend on platform default for "ls"'s ordering.

Reply:
It's [here](https://github.com/apache/camel-k/blob/059ae48ac293c6d28783cc958d0c39cc505e26fc/script/embed_resources.sh#L41) where I would add a `| sort` in between.

Reply:
not sure what appended but I see this commit

    https://github.com/apache/camel-k/commit/c2053bfe2155bb1e418457840434d8b07bde6985

Reply:
Sorry, there is already a sort at the end.

So no idea how this file was regenerated the last time.

Reply:
so looks like there are some differences between linux and osx implementations

Reply:
I don't think so as this final `| sort` works the same on linux and osx. No idea though why the resource list checked in in master is not sorted.

I suggest to merge this in (as its sorted now), and check for a PR which changes this again into a non-sorted order.